plugins/kotlin/j2k/shared/src/org/jetbrains/kotlin/nj2k/tree/visitors/JKVisitor.kt

```diff
@@ -6,165 +6,182 @@ import org.jetbrains.kotlin.nj2k.tree.*
+
+    abstract fun visitField(field: JKField)
+
+    abstract fun visitKtInitDeclaration(ktInitDeclaration: JKKtInitDeclaration)
+
+    abstract fun visitTreeRoot(treeRoot: JKTreeRoot)
+
+    abstract fun visitFile(file: JKFile)
+
+    abstract fun visitTypeElement(typeElement: JKTypeElement)
+
+    abstract fun visitBlock(block: JKBlock)
+
+    abstract fun visitInheritanceInfo(inheritanceInfo: JKInheritanceInfo)
+
+    abstract fun visitPackageDeclaration(packageDeclaration: JKPackageDeclaration)
+
+    abstract fun visitLabelEmpty(labelEmpty: JKLabelEmpty)
+
+    abstract fun visitLabelText(labelText: JKLabelText)
+
+    abstract fun visitImportStatement(importStatement: JKImportStatement)
+
+    abstract fun visitImportList(importList: JKImportList)
+
+    abstract fun visitAnnotationParameter(annotationParameter: JKAnnotationParameter)
+
+    abstract fun visitAnnotationNameParameter(annotationNameParameter: JKAnnotationNameParameter)
+
+    abstract fun visitArgument(argument: JKArgument)
+
+    abstract fun visitNamedArgument(namedArgument: JKNamedArgument)
+
+    abstract fun visitArgumentList(argumentList: JKArgumentList)
+
+    abstract fun visitTypeParameterList(typeParameterList: JKTypeParameterList)
+
+    abstract fun visitAnnotationList(annotationList: JKAnnotationList)
+
+    abstract fun visitAnnotation(annotation: JKAnnotation)
+
+    abstract fun visitTypeArgumentList(typeArgumentList: JKTypeArgumentList)
+
+    abstract fun visitNameIdentifier(nameIdentifier: JKNameIdentifier)
+
+    abstract fun visitKtWhenCase(ktWhenCase: JKKtWhenCase)
+
+    abstract fun visitKtElseWhenLabel(ktElseWhenLabel: JKKtElseWhenLabel)
+
+    abstract fun visitKtValueWhenLabel(ktValueWhenLabel: JKKtValueWhenLabel)
+
+    abstract fun visitClassBody(classBody: JKClassBody)
+
+    abstract fun visitBinaryExpression(binaryExpression: JKBinaryExpression)
+
+    abstract fun visitPrefixExpression(prefixExpression: JKPrefixExpression)
+
+    abstract fun visitPostfixExpression(postfixExpression: JKPostfixExpression)
+
+    abstract fun visitQualifiedExpression(qualifiedExpression: JKQualifiedExpression)
+
+    abstract fun visitParenthesizedExpression(parenthesizedExpression: JKParenthesizedExpression)
+
+    abstract fun visitTypeCastExpression(typeCastExpression: JKTypeCastExpression)
+
+    abstract fun visitLiteralExpression(literalExpression: JKLiteralExpression)
+
+    abstract fun visitStubExpression(stubExpression: JKStubExpression)
+
+    abstract fun visitThisExpression(thisExpression: JKThisExpression)
+
+    abstract fun visitSuperExpression(superExpression: JKSuperExpression)
+
+    abstract fun visitIfElseExpression(ifElseExpression: JKIfElseExpression)
+
+    abstract fun visitLambdaExpression(lambdaExpression: JKLambdaExpression)
+
+    abstract fun visitDelegationConstructorCall(delegationConstructorCall: JKDelegationConstructorCall)
+
+    abstract fun visitCallExpression(callExpressionImpl: JKCallExpressionImpl)
```

Comment:
It feels like a mistake that the parameter type is `JKCallExpressionImpl` instead of `JKCallExpression`.